conda/models/environment.py

```diff
@@ -160,18 +163,17 @@ def merge(cls, *configs: EnvironmentConfig) -> EnvironmentConfig:
         """
 
         # Don't try to merge if there is nothing to merge
-        if len(configs) == 0:
+        if not configs:
             return
 
         # If there is only one config, there is nothing to merge, return the lone config
         if len(configs) == 1:
             return configs[0]
 
-        # Iterate over all configs and merge them into the result
-        result = configs[0]
-        for config in configs[1:]:
-            result._merge(config)
-        return result
+        # Use reduce to merge all configs into the first one
+        return reduce(
+            lambda result, config: result._merge(config), configs[1:], configs[0]
```

Comment:
```suggestion
            lambda result, config: result._merge(config), configs[1:], initial=configs[0]
```

I was confused by the 3rd argument, I think it's clearer this way (although I prefer Sophia's original code due to my lack of familiarity with functional constructs, they are not very common in this codebase either, 4 results total).